js/vue/README.md

```diff
@@ -0,0 +1,75 @@
+
+### What to make
+
+We render our Vue components as _Single_ _File_ _Components_ (SFC). This means that all of the html, scripts and css that Vue needs to render the components properly is all wrapped up in the same place.
+
+This has a couple of distinct advantages, most notably that it become trivial to move the component because everything that you need is wrapped up in the single file together. You can read more about it on [Vues Documentation](https://vuejs.org/v2/guide/single-file-components.html)
+
+As a word of caution, it is sometimes easy to get a bit over zealous with the size of a SFC, and end up with a ton of HTML, CSS, and JavaScript that you have to try and understand at once. The cool part is that if you _do_ find that your SFC is too much to work with in one go, its easy to extract part of the file into another SFC that you can import and include, letting you build up a components of components and making it easier to understand the UI. You probably won't run into this problem for the site, but if you do then that is an excellent way to reduce complexity.
+
+### How to include it
+
+Once you have made your SFC, we need to include it in the bundle of JavaScript we make for the site using webpack. Our webpack currently builds off of the `webpack/entry.js` file, which then bundles the JavaScript and puts it into `js/bundle.js`
+
+For example, I have made the `meetupEventsVue.vue` SFC to include in the application on the homepage. I need to get that component into our build pipeline, and to do that I am going to first import it into our `entry.js`...
+
+```javascript
+//use relative path to the file from the entry.js file
+//make sure you use export default in the script of your SFC to make an export
+import MeetupEvents from '../js/components/vue/meetupEventsVue.vue';
+```
+once it is included in the `entry.js` at the top, you now need to tell JavaScript what part of the DOM is controlled by your Vue Instance. Usually, I use an `id` on an empty `div` element like so ...
+
+```html
+<div id="meetup-events">
+</div>
+```
+
+You can in theory use any css selector with vue, but I find it helpful to use id because html enforces that I don't accidentally duplicate ids and end up with multiple of the same Vue Instance.
+
+Now the last bit is to let our Vue instance take control and load it _after_ the document is ready. In this case we will use jQuery to do that, because we already have jQuery as a dependency but you can also use vanilla JavaScript to load after the page has loaded if you like.
+
+```javascript
+$( document ).ready( function() {
+  var eventsApp = new Vue({
+    el: "#meetup-events",
+    render: function(h) {
+      return h(MeetupEvents);
+    },
+  })
+})
+```
```

Comment:
Would it be appropriate to have one Vue instance for the whole page and these little features be components within that view? I'm asking mainly because I don't know Vue well enough to know the difference.

Reply:
So I would say that if the whole page is displayed by Vue, then absolutely. But in our case we probably want to have vue instances for only the areas that have interactivity with the DOM and then the rest is static HTML.

For example, if we wanted to take our mobile bootstrap button and make it a vue component, it would probably be easiest to just have Vue control exactly that button and the meetup events separately. Putting them under one vue instance doesn't provide any benefit (because they don't need to share any infomration) and now we are rendering the whole masthead and whatever static html is between those 2 in Vue for no reason.

But there isn't anything wrong with using Vue to render the whole page, but at that point you aren't really using Jekyll, you are using Vue, and I like the way that we use jekyll for most things then a sprinkling of vue for dynamic elements.